core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.ts

```diff
@@ -173,7 +222,125 @@ export class TranslationModalComponent {
+      originalElements: string[],
+      translatedElements: string[]): boolean {
+    if (originalElements.length === 0) {
+      return false;
+    }
+    const hasMatchingTranslatedElement = element => (
+      translatedElements.includes(element) && originalElements.length > 0);
+    return originalElements.some(hasMatchingTranslatedElement);
+  }
+
+  isTranslationCompleted(
+      originalElements: HTMLElement[],
+      translatedElements: HTMLElement[]): boolean {
+    const filteredOriginalElements = originalElements.filter(
+      element => element.nodeType === Node.ELEMENT_NODE);
+    const filteredTranslatedElements = translatedElements.filter(
+      element => element.nodeType === Node.ELEMENT_NODE);
+
+    if (filteredOriginalElements.length !== filteredTranslatedElements.length) {
+      return false;
+    }
+
+    for (const [i, originalElement] of filteredOriginalElements.entries()) {
+      if (originalElement.nodeName !== filteredTranslatedElements[
+        i].nodeName) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  validateTranslation(
+      textToTranslate: HTMLElement[],
+      translatedText: HTMLElement[]): TranslationError {
+    const translatedElements: ImageDetails = this.getImageAttributeTexts(
+      translatedText);
+    const originalElements: ImageDetails = this.getImageAttributeTexts(
+      textToTranslate);
+
+    const hasUncopiedImgs = !this.copiedAllElements(
+      originalElements.filePaths,
+      translatedElements.filePaths);
+    const hasDuplicateAltTexts = this.hasSomeDuplicateElements(
+      originalElements.alts,
+      translatedElements.alts);
+    const hasDuplicateDescriptions = this.hasSomeDuplicateElements(
+      originalElements.descriptions,
+      translatedElements.descriptions);
+    const hasUntranslatedElements = !(this.isTranslationCompleted(
+      textToTranslate, translatedText));
+
+    return new TranslationError(
+      hasUncopiedImgs, hasDuplicateAltTexts,
+      hasDuplicateDescriptions, hasUntranslatedElements);
+  }
+
   suggestTranslatedText(): void {
+    const originalElements = Array.from(angular.element(
+      this.textToTranslate));
+    const translatedElements = Array.from(angular.element(
+      this.activeWrittenTranslation.html));
+
+    const translationError = this.validateTranslation(
+      originalElements, translatedElements);
+
+    this.hasImgCopyError = translationError.hasUncopiedImgs;
+    this.hasImgTextError = translationError.hasDuplicateAltTexts ||
+      translationError.hasDuplicateDescriptions;
+    this.incompleteTranslationError = translationError.
+      hasUntranslatedElements;
+
+    if (this.hasImgCopyError || this.hasImgTextError ||
+      this.incompleteTranslationError || this.uploadingTranslation ||
+      this.loadingData) {
+      return;
+    }
+
+    this.hasImgCopyError = false;
+    this.triedToCopyParagraph = false;
+    this.hasImgTextError = false;
+    this.incompleteTranslationError = false;
```

Comment:
Why is this necessary? If the user fixed the validation issues and tries to submit the translation again, shouldn't these already be set to false above?

Reply:
Sorry! It is unnecessary except `hadCopyParagraphError`. When `hadCopyParagraphError` is set to be `true` it is not set as `false` unless user clicks on an item which is allowed to be copied. Hence once the user finishes the translation, `hadCopyParagraphError` is set to be `false` becaus if it is not set to be `false`, paragraph copy error will be shown if the user tries another translation while being in the same modal.